### PR TITLE
Add cooldown period to avoid immediate termination of telemetry monitor

### PR DESCRIPTION
**Pull request by ankona**
The initial telemetry monitor merge ignored a known automatic shutdown bug. The mitigation removed automatic shutdown completely.

This change solves an automatic shutdown issue when the telemetry monitor has not yet begun to monitor a task. The prior auto-shutdown condition allowed the telemetry monitor to immediately complete. 

A cooldown/timeout period has been added to avoid immediate shutdown. The telemetry monitor can now be configured to live a specific duration and avoids shutdown prior to the writing of a runtime manifest.

### CODE REVIEW

File: tests/test_telemetry_monitor.py [hunk 1]
```diff
@@ -797,3 +763,50 @@ def test_telemetry_colo(fileutils, wlmutils, coloutils, monkeypatch):
...
+        assert popen.returncode is None
+
+        # give some leeway during testing for the cooldown to get hit
+        for i in range(10):
+            if popen.poll() is not None:
+                print(f"Completed polling for telemetry shutdown after {i} attempts")
+                break
+            time.sleep(3)
+
+        assert popen.returncode is not None
```
Comment (ankona) [inline tests/test_telemetry_monitor.py:812]: should add an `assert elapsed >= cooldown` to ensure it was really honored

**Review by AlyssaCote — COMMENTED** [review]
This looks good to me! 👍

File: tests/test_telemetry_monitor.py (hunk 1 above)
Comment (ankona) [reply tests/test_telemetry_monitor.py:829]: added

File: smartsim/_core/entrypoints/telemetrymonitor.py [hunk 2]
```diff
@@ -478,50 +481,45 @@ def on_timestep(self, timestamp: int) -> None:
...
+    has_running_jobs = has_jobs or has_dbs
 
-def shutdown_when_completed(
-    observer: BaseObserver, action_handler: ManifestEventHandler
-) -> None:
-    """Inspect active and completed job queues and shutdown if all jobs are complete
+    if has_jobs:
+        logger.debug(f"telemetry monitor is monitoring {len(jobs)} jobs")
+    if has_dbs:
+        logger.debug(f"telemetry monitor is monitoring {len(jobs)} dbs")
```
Comment (al-rigazzi) [inline smartsim/_core/entrypoints/telemetrymonitor.py:512]: `jobs` -> `db_jobs` ?

File: smartsim/_core/entrypoints/telemetrymonitor.py [hunk 3]
```diff
@@ -625,6 +638,12 @@ def get_parser() -> argparse.ArgumentParser:
         help="Experiment root directory",
         required=True,
     )
+    arg_parser.add_argument(
+        "-cooldown",
+        type=int,
+        help="Default lifetime of telemetry monitor (in seconds)) before auto-shutdown",
```
Comment (al-rigazzi) [inline smartsim/_core/entrypoints/telemetrymonitor.py:644]: Two )) for only one (

**Review by al-rigazzi — CHANGES_REQUESTED** [review]
A couple of minor typos, otherwise looks good!

File: smartsim/_core/entrypoints/telemetrymonitor.py (hunk 2 above)
Comment (ankona) [reply smartsim/_core/entrypoints/telemetrymonitor.py]: great eye. fixed

File: smartsim/_core/entrypoints/telemetrymonitor.py (hunk 3 above)
Comment (ankona) [reply smartsim/_core/entrypoints/telemetrymonitor.py]: Fixed!

**Review by al-rigazzi — APPROVED** [review]
LGTM!

**Pull request merged**